lab-03_mergesort/src/main.c

```diff
@@ -0,0 +1,70 @@
+#include <stddef.h>
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+
+#include "../include/mergesort.h"
+
+int comporator_int(const void* a, const void* b){
+	return *(int*)a - *(int*)b;
+}
+
+int comporator_char(const void* a, const void* b){
+	return *(char*)a - *(char*)b;
+}
+
+int comporator_str(const void* a, const void* b){
+	return strcmp(*(char**)a, *(char**)b);
+}
+
+int main(int argc, char* argv[]){
+	size_t n = argc - 2;
+	if (strcmp(argv[1], "int") == 0){
+		int a[n];
+		for (size_t i = 0; i < n; ++i){
+			a[i] = atoi(argv[i + 2]);
```

Comment:
Для данной работы наверное не самое ценное замечание, но точно полезное в целом. Дизайн функции `atoi` в си оставляет желать лучшего: во-первых, если ей передать строку вида `123abc`, то она вернет `123`, никак не просигнализировав о том, что вся строка вообще-то числом не является. Но еще веселее ей передавать вообще не числа: `atoi("aboba") = 0`. Первый сценарий слишком сложно полноценно проверять, а вот второй - вполне реально, поэтому давай добавим такую проверки и сделаем приложение чуть боле устойчивыс к trash in:

```suggestion
			a[i] = atoi(argv[i + 2]);
			if (a[i] == 0 && strcmp(argv[i + 2],  "0") != 0) {
			    printf("Argument number %u is incorrect: '%s' is not a number", i + 1, argv[i + 2]);
```

Reply:
и как быть с функцией atoi? лучше самому ее реализовывать? или дополнительно делать проверку на мусор?